code/modules/antagonists/changeling/changeling.dm

```diff
@@ -242,7 +242,15 @@
 			return TRUE
 	return FALSE
 
-/datum/antagonist/changeling/proc/can_absorb_dna(mob/living/carbon/human/target, verbose = TRUE)
+/**
+ * Checker proc to see if we can absorb a given human target. Returns TRUE if we can, FALSE if we can't.
+ *
+ * Arguments:
+ * * target - Who we're trying to absorb
+ * * verbose - If TRUE, print a message to the changeling about why we can't absorb, if we can't
+ * * is_true_absorb - Set to TRUE for true absoprtions, which lets them override non-true absorptions, AKA DNA stings
+ */
+/datum/antagonist/changeling/proc/can_absorb_dna(mob/living/carbon/human/target, verbose = TRUE, is_true_absorb = FALSE)
```

Comment:
This change is a good start, but the point of it was to remove the need of a new parameter.

For that you can further dismember `can_be_absorbed()` in two, one for the part that doesn't depend on that extra parameter, and one for the part that does.

Instead of doing `if(foo.can_absorb_dna(bar, is_true_absorb = TRUE))` and `if(foo.can_absorb_dna(bar, is_true_absorb = FALSE))` now you'll do `if(foo.can_absorb_dna(bar) && bar.can_be_absorbed())` and `if(foo.can_absorb_dna(bar) && bar.can_be_absorbed()) && foo.has_absorbed(bar)`

Better pass the mob instead of the changeling datum into `can_be_absorbed()`, now that `can_be_absorbed()` is to be dismembered, and make that last one as a changeling datum proc.